### PR TITLE
TypeScript conversion

### PR DESCRIPTION
**Pull request by muan**
- TypeScript
- Remove UMD build
- Update test to run on multiple OSes
- Update dependencies
- Remove Travis CI config
- Add CODEOWNERS

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -3,16 +3,17 @@
...
+  "main": "dist/index.js",
+  "type": "module",
+  "module": "dist/index.js",
+  "types": "dist/index.d.ts",
   "scripts": {
     "clean": "rm -rf dist",
     "lint": "github-lint",
-    "prebuild": "npm run clean && npm run lint",
-    "build": "rollup -c && cp src/clipboard-copy-element.js.flow dist/index.esm.js.flow && cp src/clipboard-copy-element.js.flow dist/index.umd.js.flow",
+    "prebuild": "npm run clean && npm run lint && tsc",
```
Comment (koddsson) [inline package.json:13]: Do we need `--noEmit` here?

```suggestion
    "prebuild": "npm run clean && npm run lint && tsc --noEmit",
```

File: package.json [hunk 2]
```diff
@@ -21,24 +22,22 @@
...
-    "@babel/core": "^7.6.2",
-    "babel-preset-github": "^3.2.1",
+    "@github/prettier-config": "0.0.4",
     "chai": "^4.2.0",
-    "eslint": "^6.5.1",
-    "eslint-plugin-github": "^3.1.3",
-    "flow-bin": "^0.111.1",
-    "karma": "^4.3.0",
+    "eslint": "^6.8.0",
+    "eslint-plugin-github": "^3.4.1",
```
Comment (koddsson) [inline package.json:32]: I think updating to 4.0.0 here would be good.

File: package.json (hunk 1 above)
Comment (muan) [reply package.json:13]: `tsc` compiles the code to JS which `rollup` uses here:

https://github.com/github/clipboard-copy-element/blob/89277d9910e0c60cdec83d2ed6cc09e37e25e495/rollup.config.js#L4

File: package.json (hunk 2 above)
Comment (muan) [reply package.json]: Does https://github.com/github/clipboard-copy-element/pull/30/commits/f5161d84f6a6453f425d250d6fe746ba01a3508d look right to you?

Comment (muan) [reply package.json]: Looks like it is now also causing lint failures in Windows 🤯. I'm gonna break this change into a separate branch instead.

**Pull request merged**